File: apps/android/src/main/kotlin/app/pixroost/android/spike/data/FileTransfer.kt
package app.pixroost.android.spike.data

/**
 * One file on its way to the PC. [sent] is what the PC has: the offset it reported plus what went out since.
 * An attempt is one request with the rest of the file; [resumedFrom] lists offsets of the resumed ones.
 */
data class FileTransfer(
    val file: SpikeFile,
    val status: FileStatus = FileStatus.Queued,
    val sent: Long = 0,
    val hashMillis: Long? = null,
    val attempts: Int = 0,
    val resumedFrom: List<Long> = emptyList(),
    val route: UploadRoute? = null,
    val firstAttemptAt: Long? = null,
    val attemptStartOffset: Long = 0,
    val attemptStartedAt: Long = 0,
    val finishedAt: Long? = null,
    val error: String? = null,
)

File: apps/ios/Pixroost/Spike/FileTransfer.swift
import Foundation

/// One file on its way to the PC. An attempt is one upload of the rest of the file.
struct FileTransfer {
    let file: SpikeFile
    var status: TransferStatus = .queued
    var sent: Int64 = 0
    var attempts = 0
    var resumedFrom: [Int64] = []
    var firstAttemptAt: Date?
    var attemptStartedAt: Date?
    var attemptStartOffset: Int64 = 0
    var finishedAt: Date?
    var error: String?
    let deadline = Date().addingTimeInterval(SpikeConstants.retryWindowSeconds)
}
